### PR TITLE
feat: audit log table and bpmn coloring

### PR DESCRIPTION
**Pull request by ducanhpham0312**
### Linked issue

Closes ducanhpham0312/zeevision-private#150

### Description

- Audit Log table to SingleInstancePage
- Add coloring to BPMN diagram depends on the audit log state
- Fix a bug where resizing a window including bpmn diagram breaks the application
- Update sample query to only include necessary auditlog queries.
- "Incompatible" incident queries are also modified.

### Testing

- Run test bench, start the application and run the query in pgAdmin.
- Open `/instances/11` and verify that some elements in the BPMN diagram are colored (green border, red-filled, green-filled). 
  - Green border: The element has completed or have been resolved from an incident
  - Green-filled: The element is in the activating - completing phase (requires user focus)
  - Red-filled: There is an incident occur in the element (requires user focus)
  - Gray-filled: The element is terminated.
  - The arrow is always green (I cannot color the triangle though :/ ).
  - Please also comment and make suggestion on color and opacity selection etc.
- To test the color change when an incident occurs:
  - Deploy the money-loan process
  - Create an instance without any variable. The gateway in the bpmn should be red (where incident occurs)
  - Open ZSM with that instance and set the variable (e.g., debt : 100). Then switch to the incident tab and hit resolve. The gateway should now have green border color.
- The `Audit Log` table are visible after choosing the corresponding tab.
- Open SingleProcessPage or SingleInstancePage and try resizing the browser window. It should not break the app.

### CODE REVIEW

**Review by kauppie — CHANGES_REQUESTED** [review]
Code seems alright except that I don't get the BPMN coloring:
image

**Review by kauppie — CHANGES_REQUESTED** [review]
Seems alright, but coloring doesn't work as I expected. Here is an example of how ZSM and zeevision differ:

image

I expected the path to be colored through completing the instance, but it stops at the first gateway. Does this require audit log feature to be working?

I also `Throw Error` on certain task and the coloring didn't work, although the table shows the incident:

image

File: frontend/src/pages/SingleInstancePage.tsx [hunk 1]
```diff
@@ -22,15 +22,34 @@ export default function SingleInstancesPage() {
...
     jobs,
+    auditLogs,
     incidents,
   } = instance;
-
+  const getLatestLogs = (): { elementId: string; intent: string }[] => {
+    const latestLogs: Record<string, string> = {};
+    incidents?.items.forEach((incident) => {
+      latestLogs[incident.elementId] =
+        incident.state === "CREATED" ? "INCIDENT_CREATED" : "INCIDENT_RESOLVED";
```
[the remark this answers is not in the document]
Comment (ducanhpham0312) [reply frontend/src/pages/SingleInstancePage.tsx:32]: How did you perform the "Fail the job" step though? I initially thought incident only has 2 states `CREATED` and `RESOLVED`... what is the incident state when you fail the job? And how about reversing the ternary operator:
```suggestion
        incident.state === "RESOLVED" ? "INCIDENT_RESOLVED" : "INCIDENT_CREATED";
```

**Comment by ducanhpham0312** [comment]
And to answer your PR comment, yes, everything requires the audit log store API so it's not visible when you start an instance. If you create an instance that caused an incident in the progress, then you may see the coloring.

File: frontend/src/pages/SingleInstancePage.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ducanhpham0312) [reply frontend/src/pages/SingleInstancePage.tsx:32]: I tested myself and the task turns red :D 
Screenshot 2023-12-09 at 2 24 22

[the remark this answers is not in the document]
Comment (kauppie) [reply frontend/src/pages/SingleInstancePage.tsx:32]: I retried and now it works, somehow. Can you confirm that the incident table doesn't show correct information in the `Element ID`  field as it differs from ZSM?

**Comment by ducanhpham0312** [comment]
The latest commit fixed the table pagination when selecting "All". It was fixed in a previous PR, but somehow got reverted :/

File: frontend/src/components/BpmnViewer/BpmnViewer.tsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
[the remark this answers is not in the document]
Comment (kauppie) [reply frontend/src/components/BpmnViewer/BpmnViewer.tsx]: This wasn't fixed.

**Review by kauppie — APPROVED** [review]

**Pull request merged**